lib/fog/azurerm/models/compute/managed_disk.rb

```diff
@@ -46,8 +46,8 @@ def save
           merge_attributes(Fog::Compute::AzureRM::ManagedDisk.parse(disk))
         end
 
-        def destroy
-          service.delete_managed_disk(resource_group_name, name)
+        def destroy(async = false)
+          service.delete_managed_disk(resource_group_name, name, async)
```

Comment:
You need to check if the request is an asynchronous request or not and return the response accordingly. In case of a synchronous request the response will return as is. In case of an asynchronous request you will need to convert the response into a Fog::AsyncResponse object and return that. You can find how to do that [here](https://github.com/fog/fog-azure-rm/blob/master/lib/fog/azurerm/models/compute/server.rb#L117)